PDF import added with MuPDF integration

mpcgd opened the pull request:
- now compiling on the GitHub CI (ubuntu)
- Add CMake support for finding and linking MuPDF library (minimum version 1.18.0, syntax fixed) to enable PDF document rendering and page extraction.
- Includes new FindMuPDF.cmake module, debug macro for Debug builds, and updated Qt5 path handling for macOS.
- Enhances application capabilities by allowing PDF file processing with proper error handling.
- Pdf images are extracted if the page is a single image, rendered otherwise.
- Fix of the multipage thumbnail generation.
- CodeQL workflow activated for main & master branch
- tested on MacOS

trufanov-nok committed:
Update control: add libmupdf to deps

trufanov-nok committed:
Update codeql-analysis.yml

Commit:
feat(build): integrate MuPDF for PDF rendering support

Add CMake support for finding and linking MuPDF library (minimum version 1.18.0) to enable PDF document rendering and page extraction. Includes new FindMuPDF.cmake module, debug macro for Debug builds, and updated Qt5 path handling for macOS. Enhances application capabilities by allowing PDF file processing with proper error handling

Commit:
build: enhance MuPDF linking for static Ubuntu builds

- Update GitHub Actions versions in codeql-analysis.yml for improved security and compatibility
- Add additional dependencies (zlib, harfbuzz, jbig2dec, mujs, gumbo) to CI workflow
- Refine CMakeLists.txt to properly order and link MuPDF dependencies for static builds
- Modify FindMuPDF.cmake to prefer shared libraries and integrate HarfBuzz for robust text rendering on Ubuntu

Commit:
Add libeigen3-dev to CI dependencies to fix CMake error finding Eigen3

Commit:
chore: lower libmupdf-dev minimum version to 1.18

Update Build-Depends in debian/control to require libmupdf-dev >= 1.18 instead of >= 1.25, allowing builds on systems with older libmupdf versions for improved compatibility.

Commit:
Add missing MuPDF dependencies to debian/control for consistency with CI

mpcgd commented:
Now I hope it should be ok on Linux environment too, as it is building in the GitHub CI

zvezdochiot commented:
Debian stable (== 12)

Build :ok: 

Tested :no_entry: 

:warning: `Cmake` does not check for `libgumbo-dev`, `libjbig2dec0-dev`, `libmujs-dev` (new depends)!

Completely at the discretion of @noobie-iv .

Commit:
feat(build): add MuPDF dependencies for Harfbuzz, MuJS, and Gumbo

Add support for additional MuPDF libraries (Harfbuzz for font shaping, MuJS for JavaScript execution, and Gumbo for HTML parsing) by modifying CMakeLists.txt to find and conditionally link them on Unix systems. This enhances document rendering capabilities in the project.

zvezdochiot commented:
Commit https://github.com/ImageProcessing-ElectronicPublications/scantailor-deviant/pull/23/commits/8666d59929204528fb2e2de0896f0ec0df19f4cb

```less
-- The CXX compiler identification is GNU 12.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.8.1") 
-- Checking for module 'libcanberra'
--   Found libcanberra, version 0.30
-- Found Canberra: /usr/lib/x86_64-linux-gnu/libcanberra.so  
-- Found EXIV2: /usr/lib/x86_64-linux-gnu/libexiv2.so  
-- Found MuPDF: /usr/lib/libmupdf.a (found version "1.19.0") 
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libOpenGL.so   
-- Found MuPDF: /usr/lib/libmupdf.a (found suitable version "1.19.0", minimum required is "1.18.0") 
-- Found JPEG: /usr/lib/x86_64-linux-gnu/libjpeg.so (found version "62") 
-- Found OPENJPEG: /usr/lib/x86_64-linux-gnu/libopenjp2.so  
-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version "1.2.13") 
-- Found PNG: /usr/lib/x86_64-linux-gnu/libpng.so (found version "1.6.39") 
-- Found TIFF: /usr/lib/x86_64-linux-gnu/libtiff.so (found version "4.5.0")  
-- Found Eigen3: /usr/include/eigen3 (Required is at least version "2.91.0") 
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found suitable version "1.74.0", minimum required is "1.35.0")  
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- The C compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
```

noobie-iv commented:
> Completely at the discretion of @noobie-iv

Как и обещал, после релиза я по-тихому утек с проекта. На то он и лежит не в моем личном репозитории. Делайте из него что хотите, хоть чучело (с).
